src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -475,14 +521,10 @@ func (r *Cluster) validateRedpandaCoreChanges(old *Cluster) field.ErrorList {
 	}
 	var allErrs field.ErrorList
 
-	oldRequests := old.Spec.Resources.Requests.DeepCopy()
-	oldCPURequest := oldRequests.Cpu()
-	newRequests := r.Spec.Resources.Requests.DeepCopy()
-	newCPURequest := newRequests.Cpu()
+	oldCPURequest := old.Spec.Resources.RedpandaCPU()
+	newCPURequest := r.Spec.Resources.RedpandaCPU()
```

Comment:
There might be a corner case here to be revisited:

If old cpu =  new cpu = 2
but
old rp_cpu is missing (hence = 2) & new redpanda cpu = 1
we'd hit `newCores < oldCores`.

It sounds like a valid case though.

Reply:
I might be misunderstanding, the goal is:

| old Request.Cpu | old Redpanda.Cpu | new Request.Cpu() | new Redpanda.Cpu | RedpandaCPU | Result |
| ----------- | ----------- | ----------- | ----------- | ----------- | ----------- |
| 2 | nil | 2 | nil | 2| accepted | 
| 2 | 1 | 2 | 2 | 2 | accepted |
| 2 | nil | 1 | 2 |2 |  accepted |
| 2 | nil | 1 | nil |1 |  rejected |

Reply:
Makes sense (I was missing that we don't allow an smp decrease).